src/main/java/org/betonquest/betonquest/events/FolderEvent.java

```diff
@@ -120,4 +140,18 @@ private Long getInTicks(final VariableNumber timeVariable, final Profile profile
         return time;
     }
 
+    private void register() {
+        BetonQuest.getInstance().getServer().getPluginManager().registerEvents(this, BetonQuest.getInstance());
+    }
+
+    private void unregister() {
+        PlayerQuitEvent.getHandlerList().unregister(this);
+    }
+
+    @EventHandler
+    public void onPlayerQuit(final PlayerQuitEvent event) {
+        if (cancelOnLogout) {
+            cancelled.add(PlayerConverter.getID(event.getPlayer()).getProfileUUID());
+        }
+    }
```

Comment:
**Just a note:**
We might want to consider registering the listener only if `cancelOnLogout` is set, because otherwise if doesn't even need to be called, improving performance a bit.
But in practice it doesn't really matter as the `PlayerQuitEvent` usually has quite low frequency and only a single boolean check is required once it is called.